pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -130,7 +130,17 @@ func (rs *ReceiverService) GetReceiver(ctx context.Context, q models.GetReceiver
 		return nil, err
 	}
 
-	rs.decryptOrRedactSecureSettings(ctx, rcv, q.Decrypt)
+	if q.Decrypt {
+		err := rcv.Decrypt(rs.decryptor(ctx))
+		if err != nil {
+			rs.log.Warn("failed to decrypt secure settings", "name", rcv.Name, "error", err)
+		}
+	} else {
+		err := rcv.Encrypt(rs.encryptor(ctx))
```

Comment:
`Encrypt` here is not strictly necessary so I can remove if reviewers prefer. I chose to `Encrypt` here as it will fix any secure fields left in `Settings` that might have been missed when this was previously persisted.